### ISSUE TITLE
Unit Tests testFixAllScope related problem

### ISSUE THREAD

**Issue by TKharaishvili** [dotnet/roslyn-analyzers#2237]
While working on this PR - https://github.com/dotnet/roslyn-analyzers/pull/2234 several problematic unit tests were revealed, 20 to be exact, all of which have for now been silenced by passing `testFixAllScope: null`.

They all need to be examined. The 2nd commit of the PR documents all those problematic unit tests.

**Comment by Evangelink** [comment]
@sharwell Do you think this issue is still relevant with the recent move to the sdk test framework?

**Comment by sharwell** [comment]
@Evangelink Yes, these cases now show up as tests with `NumberOfFixAllIterations` set to something greater than 1. For example:

https://github.com/dotnet/roslyn-analyzers/blob/79c2638f54b9e7ef0a3954cc838ec4c13ea63c1f/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DefineAccessorsForAttributeArgumentsTests.Fixer.cs#L115

**Comment by Evangelink** [comment]
Here is the list of fixers with `NumberOfFixAllIterations` set to something greater than 1:

- DefineAccessorsForAttributeArgumentsFixer
- DoNotDirectlyAwaitATaskFixer
- ImplementStandardExceptionConstructorsFixer
- InterfaceMethodsShouldBeCallableByChildTypesFixer
- UriParametersShouldNotBeStringsFixer
- CSharpReviewUnusedParametersFixer
- BasicReviewUnusedParametersFixer
- ImportingConstructorShouldBeObsoleteCodeFixProvider
